bugbug/model.py

```diff
@@ -455,6 +456,19 @@ def train(self, importance_cutoff=0.15, limit=None):
         y_pred = self.clf.predict(X_test)
         y_pred = self.le.inverse_transform(y_pred)
 
+        # calibrating the model
+
+        # Fit isotonic regression model to the predicted probabilities
+        iso_reg = IsotonicRegression(out_of_bounds="clip")
+        iso_reg.fit(y_pred, y_test)
```

Comment:
The testing data should not be used to train the model.

Reply:
But we dont have any Validation data and the IsotonicRegression model needs to be trained on the predictions made by the XGBoost Model on the validation data

Reply:
> Using the training data to fit the IsotonicRegression model can lead to overfitting!

Yes, so we need to create validation data. Currently, we split the data into train/test. Instead, we could split it into train/validation/test.

Reply:
Great! I'll do that.

Reply:
@suhaibmujahid how much percentage of data should i have in train, val and test, is there a specific number you would recommend or should I do it as I see fit?

Reply:
30% of the training set would be ok

Reply:
i did 80% training, 10% validation and 10% test!